tapiriik/services/Setio/setio.py

```diff
@@ -0,0 +1,264 @@
+        ActivityType.Hiking: "Hike",
+        ActivityType.Running: "Run",
+        ActivityType.Walking: "Walk",
+        ActivityType.Snowboarding: "Snowboard",
+        ActivityType.Skating: "IceSkate",
+        ActivityType.CrossCountrySkiing: "NordicSki",
+        ActivityType.DownhillSkiing: "AlpineSki",
+        ActivityType.Swimming: "Swim",
+        ActivityType.Gym: "Workout",
+        ActivityType.Rowing: "Rowing",
+        ActivityType.Elliptical: "Elliptical",
+        ActivityType.RollerSkiing: "RollerSki",
+        ActivityType.StrengthTraining: "WeightTraining",
+    }
+
+    # For mapping Setio->common
+    _reverseActivityTypeMappings = {
+        "Ride": ActivityType.Cycling,
+        "VirtualRide": ActivityType.Cycling,
+        "EBikeRide": ActivityType.Cycling,
+        "MountainBiking": ActivityType.MountainBiking,
+        "Run": ActivityType.Running,
+        "Hike": ActivityType.Hiking,
+        "Walk": ActivityType.Walking,
+        "AlpineSki": ActivityType.DownhillSkiing,
+        "CrossCountrySkiing": ActivityType.CrossCountrySkiing,
+        "NordicSki": ActivityType.CrossCountrySkiing,
+        "BackcountrySki": ActivityType.DownhillSkiing,
+        "Snowboard": ActivityType.Snowboarding,
+        "Swim": ActivityType.Swimming,
+        "IceSkate": ActivityType.Skating,
+        "Workout": ActivityType.Gym,
+        "Rowing": ActivityType.Rowing,
+        "Kayaking": ActivityType.Rowing,
+        "Canoeing": ActivityType.Rowing,
+        "StandUpPaddling": ActivityType.Rowing,
+        "Elliptical": ActivityType.Elliptical,
+        "RollerSki": ActivityType.RollerSkiing,
+        "WeightTraining": ActivityType.StrengthTraining,
+    }
+
+    SupportedActivities = list(_activityTypeMappings.keys())
+
+    def WebInit(self):
+        params = {'scope':'write,view_private',
+                  'client_id':SETIO_CLIENT_ID,
+                  'response_type':'code',
+                  'redirect_uri':WEB_ROOT + reverse("oauth_return", kwargs={"service": "setio"})}
+        self.UserAuthorizationURL = \
+            "https://setio.run/oauth/authorize?" + urlencode(params)
+
+    def _apiHeaders(self, serviceRecord):
+        return {"Authorization": "access_token " + serviceRecord.Authorization["OAuthToken"]}
+
+    def RetrieveAuthorizationToken(self, req, level):
+        code = req.GET.get("code")
+        authorizationData = {"OAuthToken": code}
+        return (code, authorizationData)
+
+    def RevokeAuthorization(self, serviceRecord):
+        #  you can't revoke the tokens setio distributes :\
+        pass
+
+
+    def DownloadActivityList(self, svcRecord, exhaustive=False):
+        activities = []
+        exclusions = []
+
+        url = self.SetioDomain + "getRunsByUserId"
+        extID = svcRecord.ExternalID
+
+        #payload = "{\"userId\": \"" + extID + "\"}"
+        payload = {}
+        payload["userId"] = extID
+        payload=json.dumps(payload, ensure_ascii=False)

```

Comment:
I need the json.dumps on the post - but refactored it a bit - also I see Endomondo doing the same thing -